src/SonarScanner.MSBuild.PreProcessor/AnalysisConfigGenerator.cs

```diff
@@ -30,63 +30,47 @@ public static class AnalysisConfigGenerator
         /// Combines the various configuration options into the AnalysisConfig file
         /// used by the build and post-processor. Saves the file and returns the config instance.
         /// </summary>
-        /// <param name="localSettings">Processed local settings, including command line arguments supplied the user</param>
+        /// <param name="localSettings">Processed local settings, including command line arguments supplied the user.</param>
         /// <param name="buildSettings">Build environment settings</param>
-        /// <param name="serverProperties">Analysis properties downloaded from the SonarQube server</param>
-        /// <param name="analyzerSettings">Specifies the Roslyn analyzers to use. Can be empty</param>
+        /// <param name="additionalSettings">Additional settings generated by this Scanner. Can be empty.</param>
```

Comment:
`additionalSettings` can be empty but not null. It is used in a `foreach` and it will throw there. We should add a check and a test.